src/store/safePass.ts

```diff
@@ -0,0 +1,36 @@
+import { cgwDebugStorage } from '@/components/sidebar/DebugToggle'
+import { IS_PRODUCTION, GATEWAY_URL_PRODUCTION, GATEWAY_URL_STAGING } from '@/config/constants'
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+
+const GATEWAY_URL = IS_PRODUCTION || cgwDebugStorage.get() ? GATEWAY_URL_PRODUCTION : GATEWAY_URL_STAGING
+const GLOBAL_CAMPAIGN_IDS: Record<'1' | '11155111', string> = {
+  '11155111': 'fa9f462b-8e8c-4122-aa41-2464e919b721',
+  '1': '9ed78b8b-178d-4e25-9ef2-1517865991ee',
+}
+
+export type CampaignLeaderboardEntry = {
+  holder: string
+  position: number
+  boost: string
+  totalPoints: number
+  totalBoostedPoints: number
+}
+
+export const safePassApi = createApi({
+  reducerPath: 'safePassApi',
+  baseQuery: fetchBaseQuery({ baseUrl: GATEWAY_URL }),
+  endpoints: (builder) => ({
+    getOwnGlobalCampaignRank: builder.query<
+      CampaignLeaderboardEntry,
+      { chainId: '1' | '11155111'; safeAddress: string }
+    >({
+      query: (request) => ({
+        url: `v1/community/campaigns/${GLOBAL_CAMPAIGN_IDS[request.chainId]}/leaderboard/${request.safeAddress}`,
+      }),
+    }),
+  }),
+})
```

Comment:
I think this could be part of a general `gatewayApi` slice like we added in #4020 and #4017

Reply:
Those PRs are not merged yet. We should consolidate it into once slice when all those PRs are merged.